plugins/BEdita/Core/src/Model/Table/ApplicationsTable.php

```diff
@@ -85,19 +94,25 @@ public function buildRules(RulesChecker $rules)
     }
 
     /**
-     * Generate the api key on application creation
+     * Generate the api key on application creation.
+     *
+     * If applications is DEFAULT_APPLICATION or current invoking application and `enabled` is `false`
+     * raise an ImmutableResourceException
      *
      * @param \Cake\Event\Event $event The event dispatched
      * @param \Cake\Datasource\EntityInterface $entity The entity to save
      * @return void
      */
```

Comment:
Some doc about exception, i.e.: `@throws \BEdita\Core\Exception\ImmutableResourceException if entity is immutable`

Reply:
rigth, done